web/src/popover_menus_data.ts

```diff
@@ -248,13 +250,15 @@ export function get_topic_popover_content_context({
     const visibility_policy = user_topics.get_topic_visibility_policy(sub.stream_id, topic_name);
     const all_visibility_policies = user_topics.all_visibility_policies;
     const is_spectator = page_params.is_spectator;
+    const is_topic_empty = message_util.get_messages_in_topic(stream_id, topic_name).length === 0;
```

Comment:
This check is not correct unless we have all the messages in the local cache. I think the correct check is a bit subtle.

I think we want to call the thing that we'll compute `is_topic_definitely_empty`, and I think its value can be true if and only if all of the following are true:
- It's the currently active channel/topic pair (`narrow_state` should support checking this)
- `message_lists.current.data.empty()` is true
- `message_lists.current.data.fetch_status.has_found_oldest()` is true

(We only show empty topics in the sidebar if it's the current one)

Reply:
Thank you for the valuable feedback. I will work on this.